### PR TITLE
[receiver/sqlserverreceiver] unexport MockPerfCounterWatcher

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

File: .chloggen/sqlserverreceiver-unexport.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: receiver/sqlserverreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Unexport sqlserverreceiver `MockPerfCounterWatcher`
```
Comment (dmitryax) [inline .chloggen/sqlserverreceiver-unexport.yaml:10]: I'm sorry, it looks like I was wrong. I just checked if objects exported in `_test.go` files are actually available outside, appeared they are not. So this changelog isn't needed

**Comment by atoulme** [comment]
Tests fail with #24888

File: .chloggen/sqlserverreceiver-unexport.yaml (hunk 1 above)
Comment (atoulme) [reply .chloggen/sqlserverreceiver-unexport.yaml]: No problem, removed.

**Comment by djaglowski** [comment]
I believe https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24900 should fix the failure

**Review by djaglowski — APPROVED** [review]

**Pull request merged**